plugins/user-authenticators/ldap/src/org/apache/cloudstack/ldap/LdapConfiguration.java

```diff
@@ -36,6 +36,8 @@
 
     private static final ConfigKey<Integer> ldapPageSize = new ConfigKey<Integer>(Integer.class, "ldap.request.page.size", "Advanced", "1000",
                                                                                "page size sent to ldap server on each request to get user", true, ConfigKey.Scope.Global, 1);
+    private static final ConfigKey<String> ldapProvider = new ConfigKey<String>(String.class, "ldap.provider", "Advanced", "openldap", "ldap provider ex:openldap, microsoftad",
+                                                                                true, ConfigKey.Scope.Global, null);
```

Comment:
As part of this change please move all LDAP related config to use the ConfigKey/Configurable mechanism of publishing them.